jujugui/static/gui/src/app/app.js

```diff
@@ -854,6 +858,46 @@ YUI.add('juju-gui', function(Y) {
     },
 
     /**
+      Callback handler for the api loginWithMacaroon method which hanadles
+      the "redirection required" error message.
+
+      @method _apiLoginHandler
+      @param {Object} api The API that the user is attempting to log into.
+      @param {String} err The login error message, if any.
+      @param {Object} data The login request response data, if any.
+    */
+    _apiLoginHandler: function(api, err, data) {
+      const errorNotify = function(err) {
+        this.db.notifications.add({
+          title: 'Unable to log into model',
```

Comment:
Oh I see now that this is only used when the model connection fails, so nevermind.